src/ray/core_worker/task_manager.cc

```diff
@@ -119,10 +145,33 @@ void TaskManager::PendingTaskFailed(const TaskID &task_id, rpc::ErrorType error_
         }
       }
     }
+    RemovePlasmaSubmittedTaskReferences(spec);
     MarkPendingTaskFailed(task_id, spec, error_type);
   }
 }
 
+void TaskManager::RemoveSubmittedTaskReferences(const std::vector<ObjectID> &object_ids) {
+  std::vector<ObjectID> deleted;
+  reference_counter_->RemoveSubmittedTaskReferences(object_ids, &deleted);
+  in_memory_store_->Delete(deleted);
```

Comment:
should we delete from plasma?

Reply:
In the future we will, but for now we don't actually do object pinning/freeing from plasma